corehq/apps/geospatial/tasks.py

```diff
@@ -9,3 +26,32 @@ def geo_cases_reassignment_update_owners(domain, case_owner_updates_dict, task_k
     finally:
         celery_task_tracker = CeleryTaskTracker(task_key)
         celery_task_tracker.mark_completed()
+
+
+@serial_task('async-index-es-docs', timeout=30 * 60, queue='background_queue', ignore_result=True)
+def index_es_docs_with_location_props(domain):
+    celery_task_tracker = get_celery_task_tracker(domain, INDEX_ES_TASK_HELPER_BASE_KEY)
+    if celery_task_tracker.is_active():
+        return
+
+    geo_case_prop = get_geo_case_property(domain)
+    query = _es_case_query(domain, geo_case_prop)
+    doc_count = query.count()
+    if doc_count > MAX_GEOSPATIAL_INDEX_DOC_LIMIT:
+        celery_task_tracker.mark_as_error()
+        return
+
+    celery_task_tracker.mark_requested()
+    batch_count = get_batch_count(doc_count, DEFAULT_QUERY_LIMIT)
+    try:
+        for i in range(batch_count):
+            process_batch(
+                domain,
+                geo_case_prop,
+                case_type=None,
+                query_limit=DEFAULT_QUERY_LIMIT,
+                chunk_size=DEFAULT_CHUNK_SIZE,
+            )
+            celery_task_tracker.update_progress(current=i + 1, total=batch_count)
+    finally:
```

Comment:
Thinking of a situation when an exception occurs while processing the cases. This would mark the task as completed in the tracker. 
I think it iwould be good to handle this scenario.

Reply:
I followed the same logic as the task to async update owners. If we want to be a bit more cautious here, we could do a generic `Exception` catch, and then mark the tracker as having an error. I would need to slightly think through the tracker's current usage though, as we have no way of determining what error message to show when.

Reply:
I agree, handling an exception here more gracefully would be a good idea. Thinking through this however, I'm a little unsure about trying to store a message in the redis cache again. We would need to store the raw string in redis and then translate it when rendering it on the front-end, and I'm not entirely confident or sure whether pulling a string from the redis cache into a variable and then trying to translate it would work correctly.

@ajeety4 What do you think of extending the status system to have the ability for custom error statuses instead? For the `mark_as_error` function we could pass in an optional slug to append to the end of the "ERROR" string (e.g. "ERROR_CELERY"). Having different error statuses would allow us to know which message to show on the front-end.

Reply:
>  I'm not entirely confident or sure whether pulling a string from the redis cache into a variable and then trying to translate it would work correctly.

Great catch. You are right on this, As per django docs, 
`The caveat with using variables or computed values, as in the previous two examples, is that Django’s translation-string-detecting utility, django-admin makemessages, won’t be able to find these strings`

> What do you think of extending the status system to have the ability for custom error statuses instead? For the mark_as_error function we could pass in an optional slug to append to the end of the "ERROR" string (e.g. "ERROR_CELERY")

This is a good idea. I feel like a cleaner approach would be to use a separate key `error_slug` instead of using the `task_key` while marking it is an error. This way it keeps the choices for the `task_key` predictable while giving the flexibility to the consumer to set `error_slug` of their choice.

Reply:
Nice find! Doing message strings is definitely out then.

>  use a separate key error_slug instead of using the task_key while marking it is an error

Do you mean that the `task_key` would then only have "ACTIVE" or "ERROR" as its states and then we would keep the `error_slug` as a separate field, combining the two if the `task_key` is an error? Something like:
```
def get_status(self):
    status = self._client.get(self.task_key)
    if status == 'ERROR':
        status += f'_{self.error_slug)'
    return {
        'status': status,
        'progress': self.get_progress(),
    }
```

Reply:
Yes correct.  What do you think of returning the `error_slug` as well as shown below ? This would not then require to append or deappend a prefix

```
def get_status(self):
    status = self._client.get(self.task_key)
    return {
        'status': status,
        'error_slug': self._client.get(self.error_slug) if status == 'ERROR' else None,
        'progress': self.get_progress(),
    }


def mark_as_error(self, error_slug=None, timeout=ONE_DAY * 3):
    if error_slug:
        self._client.set(self.error_slug_key, error_slug)
    return self._client.set(self.task_key, 'ERROR', timeout=timeout)
```

Reply:
I like this idea, it would then keep it completely clean from the `task_key`. This looks like quite a small lift, so I'll implement as such.

Reply:
Addressed in [8a75544](https://github.com/dimagi/commcare-hq/pull/35126/commits/8a75544922a30270d0570a8b510ddf5ff5acf472).

Reply:
<3 Great discussion.